src/game/client/ingameui.cpp

```diff
@@ -1751,3 +1765,1227 @@ void InGameUI::Recreate_Control_Bar()
+        m_militarySubtitle->block_pos.x = m_militarySubtitle->position.x;
+        m_militarySubtitle->position.y = m_militaryCaptionPosition.y * height;
+        m_militarySubtitle->block_pos.y = m_militarySubtitle->position.y;
+        m_militarySubtitle->increment_on_frame =
+            (g_theGlobalLanguage->Get_Military_Caption_Delay_MS() * 30.0f / 1000.0f) + frame;
+        m_militarySubtitle->index = 0;
+
+        for (int i = 1; i < 4; i++) {
+            m_militarySubtitle->display_strings[i] = nullptr;
+        }
+
+        m_militarySubtitle->current_display_string = 0;
+        m_militarySubtitle->display_strings[0] = g_theDisplayStringManager->New_Display_String();
+        m_militarySubtitle->display_strings[0]->Reset();
+        m_militarySubtitle->display_strings[0]->Set_Font(g_theFontLibrary->Get_Font(m_militaryCaptionTitleFont,
+            g_theGlobalLanguage->Adjust_Font_Size(m_militaryCaptionTitlePointSize),
+            m_militaryCaptionTitleBold));
+        m_militarySubtitle->color = Make_Color(m_militaryCaptionColor.red,
+            m_militaryCaptionColor.green,
+            m_militaryCaptionColor.blue,
+            m_militaryCaptionColor.alpha);
+    }
+}
+
+void InGameUI::Remove_Military_Subtitle()
+{
+    if (m_militarySubtitle != nullptr) {
+        g_theInGameUI->Clear_Tooltips_Disabled();
+
+        for (unsigned int i = 0; i <= m_militarySubtitle->current_display_string; i++) {
+            g_theDisplayStringManager->Free_Display_String(m_militarySubtitle->display_strings[i]);
+            m_militarySubtitle->display_strings[i] = nullptr;
+        }
+
+        delete m_militarySubtitle;
+        m_militarySubtitle = nullptr;
+    }
+}
+
+void InGameUI::Display_Cant_Build_Message(LegalBuildCode code)
+{
+    switch (code) {
+        case LBC_RESTRICTED_TERRAIN:
+            g_theInGameUI->Message("GUI:CantBuildRestrictedTerrain");
+            break;
+        case LBC_NOT_FLAT_ENOUGH:
+            g_theInGameUI->Message("GUI:CantBuildNotFlatEnough");
+            break;
+        case LBC_OBJECTS_IN_THE_WAY:
+            g_theInGameUI->Message("GUI:CantBuildObjectsInTheWay");
+            break;
+        case LBC_NO_CLEAR_PATH:
+            g_theInGameUI->Message("GUI:CantBuildNoClearPath");
+            break;
+        case LBC_SHROUD:
+            g_theInGameUI->Message("GUI:CantBuildShroud");
+            break;
+        case LBC_TOO_CLOSE_TO_SUPPLIES:
+            g_theInGameUI->Message("GUI:CantBuildTooCloseToSupplies");
+            break;
+        default:
+            g_theInGameUI->Message("GUI:CantBuildThere");
+            break;
+    }
+}
+
+void InGameUI::Begin_Area_Select_Hint(GameMessage const *msg)
+{
+    m_isDragSelecting = true;
+    m_dragSelectRegion = msg->Get_Argument(0)->region;
+}
+
+void InGameUI::End_Area_Select_Hint(GameMessage const *msg)
+{
+    m_isDragSelecting = false;
+}
+
+void InGameUI::Create_Move_Hint(GameMessage const *msg)
+{
+    for (int i = 0; i < 256; i++) {

```

Comment:
MAX_MOVE_HINTS

In `W3DInGameUI` are also 2 members that could take `MAX_MOVE_HINTS`